creator-node/src/diskManager.js

```diff
@@ -0,0 +1,55 @@
+const path = require('path')
+const fs = require('fs')
+const config = require('./config')
+
+// use this set to cache existing directory paths we know we've created so we don't make extraneous file system calls
+let EXISTING_PATHS = new Set()
+
+class DiskManager {
+  /**
+   * Return the storagePath from the config
+   */
+  static getConfigStoragePath () {
+    return config.get('storagePath')
+  }
+
+  /**
+   * Construct a subdirectory path given a file CID or dir CID
+   * @dev Returns a subdirectory path with the three characters before the last character
+   *      eg QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6 will be eg /file_storage/uU/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6
+   * @param {String} fileName file CID or dir CID name
+   */
+  static computeCIDFilePath (fileName) {
+    // TODO - reject paths here
+    if (!fileName || fileName.length < 4) throw new Error(`Please pass in a valid fileName to computeCIDFilePath. Passed in ${fileName}`)
+
+    // this is the directory path that file with fileName will go into
+    const parentDirPath = path.join(this.getConfigStoragePath(), fileName.slice(-4, -1))
+    // in order to easily dev against the older and newer paths, the line below is the legacy storage path
+    // const parentDirPath = this.getConfigStoragePath()
+
+    // create the subdirectories in parentDirHash if they don't exist
+    this.ensureDirPathExists(parentDirPath)
+
+    return path.join(parentDirPath, fileName)
+  }
+
+  /**
+   * Given a directory path, this function will create the dirPath if it doesn't exist
+   * If it does exist, it will not overwrite, effectively a no-op
+   * @param {*} dirPath fs directory path to create if it does not exist
+   */
+  static ensureDirPathExists (dirPath) {
+    try {
+      if (!EXISTING_PATHS.has(dirPath)) {
```

Comment:
tbh, i dont think this optimization is really worth it since it'll only last until the service restarts and eventually we'll just saturate all the directory names possible, so we'll never be adding to the set

Reply:
yeah you're right. earlier with two characters (which is why the comment didn't have 'm' in the folder name), the chance of dirs existing was much higher since there were 3364 possible combos. i am trying to avoid calling mkdir -p on every call to `computeCIDFilePath` because that just seems unnecessary, even if it's a no-op.

Reply:
another option we could do is like on init load all the existing dir names into the set on startup but that may cause startup latency issues

Reply:
I think it's probably fine? I'd be down to ship this first without and do that optimization later

Reply:
yeah you're right we can always add later

Reply:
ended up removing this